src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -53,26 +52,24 @@ class MauiCoreInitializer : IMauiInitializeService
 			public void Initialize(IServiceProvider services)
 			{
 #if WINDOWS
-				var dispatcher = 
-					services.GetService<IDispatcher>() ??
-					MauiWinUIApplication.Current.Services.GetRequiredService<IDispatcher>();
-
-				if (!dispatcher.IsDispatchRequired)
-					SetupResources();
+				// WORKAROUND: use the MAUI dispatcher instead of the OS dispatcher to
+				// avoid crashing: https://github.com/microsoft/WindowsAppSDK/issues/2451
+				var dispatcher = services.GetRequiredService<IDispatcher>();
+				if (dispatcher.IsDispatchRequired)
+					dispatcher.Dispatch(() => SetupResources());
 				else
-					dispatcher.Dispatch(SetupResources);
+					SetupResources();
```

Comment:
An attempt to just avoid all issues and use the OS dispatcher assigned to the resource dictionary failed because of: https://github.com/microsoft/WindowsAppSDK/issues/2451